OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -334,7 +334,7 @@ bool HandleMiddleClick(ProductionItem item, ProductionIcon icon, int handleCount
 		bool HandleEvent(ProductionIcon icon, MouseButton btn, Modifiers modifiers)
 		{
 			var startCount = modifiers.HasModifier(Modifiers.Shift) ? 5 : 1;
-			var cancelCount = modifiers.HasModifier(Modifiers.Ctrl) ? CurrentQueue.QueueLength : startCount;
+			var cancelCount = modifiers.HasModifier(Modifiers.Ctrl) ? CurrentQueue.AllQueued().Count() : startCount;
```

Comment:
`.Count()` is slow. Can we instead try casting this to a collection and using `.Count`?